SalesforceCommerceCloud/pwa-kit
# [Product Tile] Promo callout on the Product Details page (@W-15784291@)

This PR makes sure that the promotion callout message on Product Details page is rendered correctly. We wanted to make sure that it's consistent when you navigate from a PLP to PDP. Unfortunately due to API inconsistency, it's not as consistent as it can be.

Most of the changes are tests, which aim to:
- test different product types
- and to also document the API inconsistency

Related PR: #1786 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

1. Navigate to this PLP http://localhost:3000/global/en-GB/search?q=roll%20sleeve%20blouse
2. Note the promo callout "10$ off roll sleeve blouse" and its promo price of 34.16
3. Click on the tile image to navigate to the PDP
4. Note the different callout message "Buy one Long Center Seam Skirt and get 2 tops" (and the price is 44.16 instead)
5. This is because of the API consistency, where the getProduct call is missing `productPromotions` in the variants.
6. However, if you then select the size S, the promo callout will be correct.
7. Try selecting other sizes, and you'll see the promo callout updated accordingly.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
